acme/acme/errors.py

```diff
@@ -11,26 +11,62 @@ class SchemaValidationError(jose_errors.DeserializationError):
 
 
 class ClientError(Error):
+    """Network error or unexpected client response."""
+    def __init__(self, response, *args, **kwargs):
+        """
+        :param requests.Response details: The response object that caused the
```

Comment:
Move docstring to `ClientError.__doc__` (using `ivar`), rename `details` to `response`